pieceio/pieceio.go

```diff
@@ -48,33 +42,29 @@ func (pio *pieceIO) GeneratePieceCommitment(payloadCid cid.Cid, selector ipld.No
 	err = pio.carIO.WriteCar(context.Background(), pio.bs, payloadCid, selector, f)
 	if err != nil {
 		cleanup()
-		return nil, nil, err
+		return nil, nil, 0, err
 	}
-	size := f.Size()
-	pieceSize := uint64(size)
-	paddedSize := pio.padReader.PaddedSize(pieceSize)
-	remaining := paddedSize - pieceSize
-	padbuf := make([]byte, remaining)
-	padded, err := f.Write(padbuf)
+	pieceSize := uint64(f.Size())
+	_, err = f.Seek(0, io.SeekStart)
 	if err != nil {
 		cleanup()
-		return nil, nil, err
-	}
-	if uint64(padded) != remaining {
-		cleanup()
-		return nil, nil, fmt.Errorf("wrote %d byte of padding while expecting %d to be written", padded, remaining)
+		return nil, nil, 0, err
 	}
-	_, err = f.Seek(0, io.SeekStart)
+	commitment, paddedSize, err := GeneratePieceCommitment(f, pieceSize)
```

Comment:
I see that you've fseek'ed the file back to 0 after the call to `pio.carIO.WriteCar`. I think it would be good manners to do the same after `GeneratePieceCommitment`.

Reply:
(if you stick with the fseekable approach)